packages/block-library/src/embed/test/index.native.js

```diff
@@ -37,6 +37,16 @@ jest.mock( 'react-native-modal', () => {
 	const mockComponent = require( 'react-native/jest/mockComponent' );
 	return mockComponent( 'react-native-modal' );
 } );
+
+// Mock debounce to prevent potentially belated state updates.
+jest.mock( 'lodash', () => ( {
+	...jest.requireActual( 'lodash' ),
+	debounce: ( fn ) => {
+		fn.cancel = jest.fn();
+		return fn;
+	},
+} ) );
```

Comment:
Another approach I tried was using the Jest timer mocks ([reference](https://jestjs.io/docs/timer-mocks)) to assure that the debounce timer was run after changing the text, however, I found conflicts with the `waitFor` helper function as it also uses timers.

If we tackle this issue in the future, I'd like to highlight an [already solved issue](https://github.com/callstack/react-native-testing-library/issues/506) opened by @dcalhoun in the React Native testing library, that will most likely help us to address the conflict, as it's already solved for the original `waitFor` function in the library.

Reply:
> For these tests I think we could stick with the approach of mocking the `debounce` function, as that logic is not intended to be covered there, nevertheless, if we tackle this issue in the future, I'd like to highlight an [already solved issue](https://github.com/callstack/react-native-testing-library/issues/506) opened by @dcalhoun in the React Native testing library, that will most likely help us to address the conflict, as it's already solved for the original `waitFor` function in the library.

I think mocking `debounce` is a good workaround for the issue today. Long-term, I think we should attempt to test the components _including the `debounce`_ usage via Jest timer mocks like you mentioned. Hopefully we can one day replace our custom `waitFor` with RNTL's default `waitFor` implementation once https://github.com/callstack/react-native-testing-library/issues/379 is resolved. WDYT?

Reply:
Agreed, once the https://github.com/callstack/react-native-testing-library/issues/379 is solved we could replace the `waitFor` function, and probably we won't even need it as we could safely use the `findBy*` and `queryBy*` query functions 🚀 .